vscode/vscode/vscode.ml

```diff
@@ -2107,10 +2171,57 @@ module Progress = struct
   include [%js: val report : t -> value:value -> unit [@@js.call]]
 end
 
+module TextDocumentContentChangeEvent = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val range : t -> Range.t [@@js.get]
+
+    val rangeLength : t -> int [@@js.get]
+
+    val rangeOffset : t -> int [@@js.get]
+
+    val text : t -> string [@@js.get]]
+end
+
+module TextDocumentChangeEvent = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val contentChanges : t -> TextDocumentContentChangeEvent.t list [@@js.get]
+
+    val document : t -> TextDocument.t [@@js.get]]
+end
+
+module TextDocumentContentProvider = struct
+  include Interface.Make ()
+
+  module OnDidChange = Event.Make (Uri)
+
+  include
+    [%js:
+    val onDidChange : t -> OnDidChange.t [@@js.get]
+
+    val provideTextDocumentContent :
+      t -> uri:Uri.t -> token:CancellationToken.t -> string ProviderResult.t
+      [@@js.call]
+
+    val create :
+         onDidChange:OnDidChange.t
```

Comment:
Since `onDidChange` can be undefined, this should be `?onDidChange:OnDidChange.t`. Add a `unit` parameter at the end to support the optional argument.

Reply:
Done in 3c87042314c6be0e944087cc3cfc14c13ef9b642